search/search_repository.go

```diff
@@ -0,0 +1,88 @@
+package search
+
+import (
+	"golang.org/x/net/context"
+
+	"log"
+	"strconv"
+
+	"github.com/almighty/almighty-core/app"
+	"github.com/almighty/almighty-core/models"
+)
+
+const (
+	fulltextPsqlQuery = `select * from work_items WHERE to_tsvector('english', id::text || ' ' || fields::text) @@ to_tsquery($1)`
+)
+
+// GormSearchRepository provides a Gorm based repository
+type GormSearchRepository struct {
+	ts  *models.GormTransactionSupport
+	wir *models.GormWorkItemTypeRepository
+}
+
+// NewGormSearchRepository creates a new search repository
+func NewGormSearchRepository(ts *models.GormTransactionSupport, wir *models.GormWorkItemTypeRepository) *GormSearchRepository {
+	return &GormSearchRepository{ts, wir}
+}
+
+func generateSearchQuery(q string) (string, error) {
+	return q, nil
+}
+
+func convertFromModel(wiType models.WorkItemType, workItem models.WorkItem) (*app.WorkItem, error) {
```

Comment:
Opened PR for the same: https://github.com/almighty/almighty-core/pull/406/files